### ISSUE TITLE
[flang] Incorrect diagnostic on component initialization using intrinsic function

### ISSUE THREAD

**Issue by DanielCChen** [llvm/llvm-project#123090]
Consider the following code:
```
  PROGRAM main
  IMPLICIT NONE
    TYPE :: DT
      CHARACTER*(INT(ACOSD(0.5)))  :: C=CHAR(INT(ACOSD(0.5)))
    END TYPE
  end
```

Flang currently issues an error as:
```
error: Semantic errors in a7.f
./a7.f:4:41: error: Must be a constant value
        CHARACTER*(INT(ACOSD(0.5)))  :: C=CHAR(INT(ACOSD(0.5)))
                                          ^^^^^^^^^^^^^^^^^^^^^
```

This seems wrong to me. All the intrinsic functions used in initialization can be folded at the compile time.
All of ifort, gfortran and XLF compile the above code successfully.

**Comment by klausler** [comment]
The degree-based variants of trigonometric functions, like `ACOSD`, are not yet supported as intrinsic procedures.

**Comment by DanielCChen** [comment]
> The degree-based variants of trigonometric functions, like `ACOSD`, are not yet supported as intrinsic procedures.

I see.  Can I keep this issue open for my tracking purpose if that is OK?

**Comment by klausler** [comment]
> > The degree-based variants of trigonometric functions, like `ACOSD`, are not yet supported as intrinsic procedures.
> 
> I see. Can I keep this issue open for my tracking purpose if that is OK?

Of course, but please update its title so that it's more clear what needs to be done.  Thanks!
